src/tests/ftest/osa/osa_online_reintegration.py

```diff
@@ -0,0 +1,247 @@
+from test_utils_pool import TestPool
+from ior_utils import IorCommand
+from job_manager_utils import Mpirun
+from write_host_file import write_host_file
+from command_utils import CommandFailure
+from mpio_utils import MpioUtils
+
+try:
+    # python 3.x
+    import queue as queue
+except ImportError:
+    # python 2.7
+    import Queue as queue
+
+
+class OSAOnlineReintegration(TestWithServers):
+    # pylint: disable=too-many-ancestors
+    """
+    Test Class Description: This test runs
+    daos_server Online reintegration test cases.
+
+    :avocado: recursive
+    """
+    def setUp(self):
+        """Set up for test case."""
+        super(OSAOnlineReintegration, self).setUp()
+        self.dmg_command = self.get_dmg_command()
+        self.no_of_dkeys = self.params.get("no_of_dkeys", '/run/dkeys/*')
+        self.no_of_akeys = self.params.get("no_of_akeys", '/run/akeys/*')
+        self.record_length = self.params.get("length", '/run/record/*')
+        self.ior_flags = self.params.get("ior_flags", '/run/ior/iorflags/*')
+        self.ior_apis = self.params.get("ior_api", '/run/ior/iorflags/*')
+        self.ior_test_sequence = self.params.get("ior_test_sequence",
+                                                 '/run/ior/iorflags/*')
+        self.ior_daos_oclass = self.params.get("obj_class",
+                                               '/run/ior/iorflags/*')
+        # Recreate the client hostfile without slots defined
+        self.hostfile_clients = write_host_file(
+            self.hostlist_clients, self.workdir, None)
+        self.pool = None
+        self.out_queue = queue.Queue()
+
+    @fail_on(CommandFailure)
+    def get_pool_leader(self):
+        """Get the pool leader
+           Returns : int (pool_leader)
+        """
+        out = []
+        kwargs = {"pool": self.pool.uuid}
+        out = self.dmg_command.get_output("pool_query", **kwargs)
+        return int(out[0][3])
+
+    @fail_on(CommandFailure)
+    def get_pool_version(self):
+        """Get the pool version
+           Returns : int (pool_version_value)
+        """
+        out = []
+        kwargs = {"pool": self.pool.uuid}
+        out = self.dmg_command.get_output("pool_query", **kwargs)
+        return int(out[0][4])
+
+    def ior_thread(self, pool, oclass, api, test, flags, results):
+        """Start threads and wait until all threads are finished.
+        Args:
+            pool (object): pool handle
+            oclass (str): IOR object class
+            API (str): IOR API
+            test (list): IOR test sequence
+            flags (str): IOR flags
+            results (queue): queue for returning thread results
+
+        Returns:
+            None
+        """
+        processes = self.params.get("slots", "/run/ior/clientslots/*")
+        container_info = {}
+        mpio_util = MpioUtils()
+        if mpio_util.mpich_installed(self.hostlist_clients) is False:
+            self.fail("Exiting Test: Mpich not installed")
```

Comment:
Is it possible to list the hosts that does not have Mpich ?

Reply:
mpich_installed checks and reports the first host... Added the necessary changes. Thanks.